### ISSUE TITLE
can not access next cloud deployment

### ISSUE THREAD

**Issue by khaledyoussef24** [threefoldtech/tfgrid-sdk-ts#1047]
after deploying next cloud and setting dns to the ip it gives site is not reachable
![image_2023-08-24_16-27-28](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/6efd4e1d-1048-4bd4-a39e-c58497b6f5c5)

digging shows it points to the exact ip
![image_2023-08-24_16-26-45](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/1eb8142a-eb04-45ee-993b-58e71972b31e)
![Screenshot from 2023-08-24 16-22-57](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/2d9c07cd-c923-4063-b477-ce92227a37b5)
![Screenshot from 2023-08-24 16-23-02](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/60ba45dd-4377-43eb-ab9e-5b5e3c1a9caf)

**Comment by khaledyoussef24** [comment]
the same issue is on devnet and qa

**Comment by Mik-TF** [comment]
@khaledyoussef24 
Thanks for creating the issue.

Did you wait a couple of minutes and retry? Or perhaps check with a dns checker to see if the propagation was 100% completed? I had this error "Site can't be reached" too when deploying and it resolved after a couple of minutes after dns propagation.

In the docs I wrote some info on dns checker so users can be warned that they might need to wait.

**Comment by Mik-TF** [comment]
@khaledyoussef24 

Just to be sure, as per the notice (see below) after deployment, you need to click on the Actions button **Open Nextcloud** that uses the url "domain.tld:8443" to set the Nextcloud instance.

(Notice: Successfully deployed a Nextcloud AIO instance. Make sure to add a DNS A record (Host: "@", Value: <Public IPv4 Address>) to your domain. After DNS propagation, under Actions, click Open Nextcloud.)

I adjusted the documentation to make this as clear as possible.

Once the installation is done, users can use "domain.tld" without :8443 to connect to their admin login page.

**Comment by Mik-TF** [comment]
I did a quick test and it works after full propagation with :8443 port.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/6f55f004-20c7-455d-a500-3b34714fc176)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/9654c61e-d825-46a6-95a0-f1d873f53ab7)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/6d6d6cdf-7695-4622-8e74-fd38ae50bb41)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/9fd29df4-f05d-4616-aab0-55afb213e823)

***

But to be clear, it won't work with threefold.pro without :8443 until the user sets the containers.

Before setting the containers:

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/cc34903d-6de6-4cd9-9c09-0ac5824fe50b)

Then you link the domain and start the containers

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/fbd8ee41-3e5a-472b-b7a7-451fb5ea300d)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/7af12156-562d-40eb-9d72-d74338774ee8)

After setting the containers you can access threefold.pro/login to sign into nextcloud. So at this point, writing threefold.pro as a URL will lead to threefold.pro/login

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/70ca3093-4e17-48f7-aa18-03586519beaf)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/da5f5d08-57d1-44d3-8396-65da771f9c4b)

Let me know what you think.
I can put all these images in the documentation and explain the whole process if it would make this ultra clear.

**Comment by Mik-TF** [comment]
I updated the docs to take into account this issue. The steps are all laid out clearly now. [See here](https://github.com/threefoldtech/info_grid/blob/5af21d44320bcf7ae94bdd5c972936649c83aa8a/src/playground/nextcloud.md)

We will be able to close this issue once the nextcloud docs is published ([PR](https://github.com/threefoldtech/info_grid/pull/240) is merged).

**Comment by Mik-TF** [comment]
The information is now clear to get the deployment running. I will close this issue.

Note that we will be updating the nextcloud weblet with gateway support. We will publish the docs and the new weblet when the gateway version is ready.

**Comment by khaledyoussef24** [comment]
i still can not deploy the solution
i tried to deploy on devnet and qa i still get the same errors
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/4f0923ed-66e2-4d27-8154-aa0d3c2263e3)
even the machine is not reachable
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/6436faa3-064b-40d7-8df3-68c3344176fd)
it requires a password

**Comment by Mik-TF** [comment]
@khaledyoussef24 
Thanks for the work.

1. URL access: After deploying the VM with nextcloud weblet, you need to access the url with :8443 at the end (the Actions button "Open Nextcloud" does it automatically)

e.g.: khalednextgded.gridtest.xyz:8443
khalednextgded.gridtest.xyz without :8443 won't work.

I tested on test and dev net, and it works (with :8443).

2. Password: I think you had a different SSH public key in the playground for dev net, so it asked for a password. With the proper SSH pub key in Profile manager, it shouldn't ask for a password.

I tested with wrong pub key, and it asked for a password. I tested with correct pub key and it worked (doesn't ask for a password) and could ssh into the VM.

**Comment by Mik-TF** [comment]
Note: the new version of the Nextcloud weblet should be clearer when it comes to opening the URL with :8443. A clear message will be shown after deployment.

**Comment by khaledyoussef24** [comment]
khalednextgded.gridtest.xyz:8443
even after trying the port it can not be reached
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/110984055/6858639b-cbfa-4fbb-b1e2-c3d78dfdca63)

I think this part should be documented 

the ssh key is set right i even made deployments for vms and they are accessible easily so is there any other solution?

**Comment by Mik-TF** [comment]
This is strange. I am not sure how to troubleshoot it since it doesn't happen on my end.

The updated Nextcloud weblet with gateway will soon be available in a few days, perhaps we can test it then and see if this issue persists.

**Comment by khaledyoussef24** [comment]
@Mik-TF is there any updates and will the gateway update will fix the issue ?

**Comment by Mik-TF** [comment]
Sorry for the delay in this issue, the gateway version took lots of testing. 

The [nextcloud gateway update](https://github.com/threefoldtech/tfgrid-sdk-ts/tree/development_ncgateway_tls) should be published as a PR this week. It might fix the issue. Matter of fact, the two issues you mention (ssh not working and site can't be reached) never happened in our testings, so it might not be an issue in the new version.

I will keep you posted on how it unfolds. Thank you for your time by the way.

**Comment by Mik-TF** [comment]
@khaledyoussef24 

If I may add: the original issue isn't about SSH issue, but about DNS propagation and "site can't be reached".

Since we cover this in the [nextcloud docs](https://manual.grid.tf/playground/nextcloud.html#dns-propagation), I think we could consider this issue to be fixed.

We could test SSH in the new weblet version and make a new issue for this explicitly if it is still happening.

What do you think?
